### PR TITLE
[Keyboard] BM68HSRGB/rev2

### PR DESCRIPTION
**Pull request by peepeetee**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/kprepublic/bm68hsrgb/rev2/config.h [hunk 1]
```diff
@@ -0,0 +1,56 @@
+
```
Comment (drashna) [inline keyboards/kprepublic/bm68hsrgb/rev2/config.h:1]: Could you add a GPL2+ compatible license header here? 

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

Or 

https://github.com/qmk/qmk_firmware/blob/6b299fab6e6a3143c061b9ea005253361fcd524b/data/templates/base/keyboard.c#L1-L15

File: keyboards/kprepublic/bm68hsrgb/rev2/config.h [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+/* Debounce reduces chatter (unintended double-presses) - set 0 if debouncing is not needed */
+#define DEBOUNCE 5
+/* disable these deprecated features by default */
+#define NO_ACTION_MACRO
+#define NO_ACTION_FUNCTION
+#define RGBLIGHT_LIMIT_VAL 150
+#define RGBLED_NUM          6
+#define RGB_DI_PIN          E2
+#define RGBLIGHT_ANIMATIONS
```
Comment (drashna) [inline keyboards/kprepublic/bm68hsrgb/rev2/config.h:32]: Could you explicitly define which animations are enabled? 
```suggestion
#define RGBLIGHT_EFFECT_BREATHING
#define RGBLIGHT_EFFECT_RAINBOW_MOOD
#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
#define RGBLIGHT_EFFECT_SNAKE
#define RGBLIGHT_EFFECT_KNIGHT
#define RGBLIGHT_EFFECT_CHRISTMAS
#define RGBLIGHT_EFFECT_STATIC_GRADIENT
#define RGBLIGHT_EFFECT_RGB_TEST
#define RGBLIGHT_EFFECT_ALTERNATING
#define RGBLIGHT_EFFECT_TWINKLE
```

File: keyboards/kprepublic/bm68hsrgb/rev2/config.h [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+#define RGBLIGHT_SAT_STEP   8
+#define RGBLIGHT_VAL_STEP   8
+
+#define RGB_DISABLE_AFTER_TIMEOUT 0 // number of ticks to wait until disabling effects
+#define RGB_DISABLE_WHEN_USB_SUSPENDED true // turn off effects when suspended
+#define RGB_MATRIX_KEYPRESSES
+#define DISABLE_RGB_MATRIX_BAND_PINWHEEL_SAT
+#define DISABLE_RGB_MATRIX_BAND_SPIRAL_SAT
+#define DISABLE_RGB_MATRIX_RAINDROPS
+#define DISABLE_RGB_MATRIX_JELLYBEAN_RAINDROPS
```
Comment (drashna) [inline keyboards/kprepublic/bm68hsrgb/rev2/config.h:44]: Deprecated. 

Please use the enable defines.

File: keyboards/kprepublic/bm68hsrgb/rev2/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,88 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/kprepublic/bm68hsrgb/rev2/keymaps/default/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

Or 

https://github.com/qmk/qmk_firmware/blob/6b299fab6e6a3143c061b9ea005253361fcd524b/data/templates/base/keyboard.c#L1-L15

File: keyboards/kprepublic/bm68hsrgb/rev2/keymaps/via/keymap.c [hunk 5]
```diff
@@ -0,0 +1,88 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/kprepublic/bm68hsrgb/rev2/keymaps/via/keymap.c:1]: Could you explicitly define which animations are enabled? 
```suggestion
#define RGBLIGHT_EFFECT_BREATHING
#define RGBLIGHT_EFFECT_RAINBOW_MOOD
#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
#define RGBLIGHT_EFFECT_SNAKE
#define RGBLIGHT_EFFECT_KNIGHT
#define RGBLIGHT_EFFECT_CHRISTMAS
#define RGBLIGHT_EFFECT_STATIC_GRADIENT
#define RGBLIGHT_EFFECT_RGB_TEST
#define RGBLIGHT_EFFECT_ALTERNATING
#define RGBLIGHT_EFFECT_TWINKLE
```

File: keyboards/kprepublic/bm68hsrgb/rev2/rev2.c [hunk 6]
```diff
@@ -0,0 +1,110 @@
+
```
Comment (drashna) [inline keyboards/kprepublic/bm68hsrgb/rev2/rev2.c:1]: Could you add a GPL2+ compatible license header here? 

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

Or 

https://github.com/qmk/qmk_firmware/blob/6b299fab6e6a3143c061b9ea005253361fcd524b/data/templates/base/keyboard.c#L1-L15

File: keyboards/kprepublic/bm68hsrgb/rev2/rev2.h [hunk 7]
```diff
@@ -0,0 +1,20 @@
+
```
Comment (drashna) [inline keyboards/kprepublic/bm68hsrgb/rev2/rev2.h:1]: Could you add a GPL2+ compatible license header here? 

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

Or 

https://github.com/qmk/qmk_firmware/blob/6b299fab6e6a3143c061b9ea005253361fcd524b/data/templates/base/keyboard.c#L1-L15

File: keyboards/kprepublic/bm68hsrgb/rev2/rules.mk [hunk 8]
```diff
@@ -0,0 +1,25 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = atmel-dfu
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = lite        # Virtual DIP switch configuration
```
Comment (drashna) [inline keyboards/kprepublic/bm68hsrgb/rev2/rules.mk:10]: There is no "full" bootmagic anymore, only the "lite" option, which can be turned on with just "yes".
```suggestion
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
```

File: keyboards/kprepublic/bm68hsrgb/rev2/rules.mk [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+EXTRAKEY_ENABLE = yes          # Audio control and System control
+CONSOLE_ENABLE = no            # Console for debug
+COMMAND_ENABLE = no            # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no          # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes              # USB Nkey Rollover
+BACKLIGHT_ENABLE = no          # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes           # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no          # Enable Bluetooth
```
Comment (drashna) [inline keyboards/kprepublic/bm68hsrgb/rev2/rules.mk:21]:
```suggestion
NKRO_ENABLE = yes              # Enable N-Key Rollover
BACKLIGHT_ENABLE = no          # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = yes          # Enable keyboard RGB underglow
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/kprepublic/bm68hsrgb/rev2/rules.mk [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = no          # Mouse keys
+EXTRAKEY_ENABLE = yes          # Audio control and System control
+CONSOLE_ENABLE = no            # Console for debug
+COMMAND_ENABLE = no            # Commands for debug and configuration
+NKRO_ENABLE = yes              # Enable N-Key Rollover
+BACKLIGHT_ENABLE = no          # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes          # Enable keyboard RGB underglow
+AUDIO_ENABLE = no              # Audio output
+RGB_MATRIX_ENABLE = yes        # Use RGB matrix
```
Comment (fauxpark) [inline keyboards/kprepublic/bm68hsrgb/rev2/rules.mk:19]:
```suggestion
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
MOUSEKEY_ENABLE = no        # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
NKRO_ENABLE = yes           # Enable N-Key Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
AUDIO_ENABLE = no           # Audio output
RGB_MATRIX_ENABLE = yes
```

File: keyboards/kprepublic/bm68hsrgb/rev2/keymaps/via/keymap.c [hunk 11]
```diff
@@ -0,0 +1,103 @@
...
+enum my_keycodes {
+    RMT = SAFE_RANGE,
+    RMS,
+    RMIH,
+    RMDH,
+    RMIS,
+    RMDS,
+    RMIV,
+    RMDV
```
Comment (fauxpark) [inline keyboards/kprepublic/bm68hsrgb/rev2/keymaps/via/keymap.c:25]:
```suggestion
    RM_TOGG = SAFE_RANGE,
    RM_MOD,
    RM_HUI,
    RM_HUD,
    RM_SAI,
    RM_SAD,
    RM_VAI,
    RM_VAD
```

File: keyboards/kprepublic/bm68hsrgb/rev2/keymaps/via/keymap.c [hunk 12]
```diff
@@ -0,0 +1,103 @@
...
+      if (record->event.pressed)
+      {rgb_matrix_decrease_sat();
+      }
+      return false;
+      case RMIV:
+      if (record->event.pressed)
+      {rgb_matrix_increase_val();
+      }
+      return false;
+      case RMDV:
```
Comment (fauxpark) [inline keyboards/kprepublic/bm68hsrgb/rev2/keymaps/via/keymap.c:64]:
```suggestion
    case RM_TOGG:
      if (record->event.pressed)
      {rgb_matrix_toggle();
      }
      return false;
    case RM_MOD:
      if (record->event.pressed)
      {rgb_matrix_step();
      }
      return false;
      case RM_HUI:
      if (record->event.pressed)
      {rgb_matrix_increase_hue();
      }
      return false;
      case RM_HUD:
      if (record->event.pressed)
      {rgb_matrix_decrease_hue();
      }
      return false;
      case RM_SAI:
      if (record->event.pressed)
      {rgb_matrix_increase_sat();
      }
      return false;
      case RM_SAD:
      if (record->event.pressed)
      {rgb_matrix_decrease_sat();
      }
      return false;
      case RM_VAI:
      if (record->event.pressed)
      {rgb_matrix_increase_val();
      }
      return false;
      case RM_VAD:
```

File: keyboards/kprepublic/bm68hsrgb/rev2/keymaps/via/keymap.c [hunk 13]
```diff
@@ -0,0 +1,103 @@
...
+        KC_TAB,         KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS, KC_PGUP,
+        KC_CAPS,        KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,          KC_ENT,  KC_PGDN,
+        KC_LSFT,                 KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT, KC_UP,   KC_END,
+        KC_LCTL,        KC_LGUI, KC_LALT,                            KC_SPC,                    KC_RALT, MO(1),   KC_RCTL, KC_LEFT, KC_DOWN, KC_RGHT
+    ),
+    [1] = LAYOUT_65_ansi(
+        KC_GESC,        KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_DEL,  RESET,
+        _______,        RGB_TOG, RGB_MOD, RGB_HUI, RGB_HUD, RGB_SAI, RGB_SAD, RGB_VAI, RGB_VAD, RGB_SPI, RGB_SPD, _______, _______, _______, _______,
+        KC_CAPS,        RMT,     RMS,     RMIH,    RMDH,    RMIS,    RMDS,    RMIV,    RMDV,    _______, _______,  _______,         _______, _______,
```
Comment (fauxpark) [inline keyboards/kprepublic/bm68hsrgb/rev2/keymaps/via/keymap.c:85]:
```suggestion
        KC_CAPS,        RM_TOGG, RM_MOD,  RM_HUI,  RM_HUD,  RM_SAI,  RM_SAD,  RM_VAI,  RM_VAD,  _______, _______,  _______,         _______, _______,
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**